ibm/data_source_ibm_database.go

```diff
@@ -0,0 +1,546 @@
+package ibm
+
+import (
+	"fmt"
+	"github.com/IBM-Cloud/bluemix-go/bmxerror"
+	"github.com/IBM-Cloud/bluemix-go/models"
+
+	"github.com/IBM-Cloud/bluemix-go/api/icd/icdv4"
+	"github.com/IBM-Cloud/bluemix-go/api/resource/resourcev1/controller"
+	"github.com/IBM-Cloud/bluemix-go/api/resource/resourcev1/management"
+
+	"github.com/hashicorp/terraform/flatmap"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func dataSourceIBMDatabaseInstance() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceIBMDatabaseInstanceRead,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Description: "Resource instance name for example, my Database instance",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+
+			"resource_group_id": {
```

Comment:
resource_group_id can be optional...If not provided we shd take default resource grp...the code below already implements..so it can be optional

Reply:
Agreed optional. Changed.